internal/controllers/clustertemplate_controller_test.go

```diff
@@ -1352,4 +1546,279 @@ var _ = Describe("validateSchemaWithoutHWTemplate", func() {
+				Name:      configmapName,
+				Namespace: namespace,
+			},
+			Data: map[string]string{
+				utils.UpgradeDefaultsConfigmapKey: `
+ibuSpec:
+  seedImageRef:
+    image: "quay.io/openshift-release-dev/ocp-release"
+    version: "4.18.0"
+  oadpContent:
+  - name: "oadp-backup"
+    namespace: "openshift-adp"
+plan:
+- actions: ["Prep"]
+- actions: ["Upgrade"]`,
+			},
+		}
+		Expect(c.Create(ctx, cm)).To(Succeed())
+
+		err := t.validateUpgradeDefaultsConfigmap(ctx, c, configmapName, namespace)
+		Expect(err).To(HaveOccurred())
+		Expect(utils.IsInputError(err)).To(BeTrue())
+		Expect(err.Error()).To(ContainSubstring("The ClusterTemplate spec.release (4.17.0) does not match the seedImageRef version (4.18.0) from the upgrade configmap"))
+	})
+
+	It("should successfully validate when IBGU spec is valid for dry-run", func() {
+		// Create a ConfigMap with a valid IBGU spec
+		// Note: The fake client doesn't perform the same validation as real K8s API server,
+		// so we test the successful path where dry-run validation passes
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      configmapName,
+				Namespace: namespace,
+			},
+			Data: map[string]string{
+				utils.UpgradeDefaultsConfigmapKey: `
+ibuSpec:
+  seedImageRef:
+    image: "quay.io/openshift-release-dev/ocp-release"
+    version: "4.17.0"
+  stage: "Idle"
+plan:
+- actions: ["Prep"]
+- actions: ["Upgrade"]`,
+			},
+		}
+		Expect(c.Create(ctx, cm)).To(Succeed())
+
+		err := t.validateUpgradeDefaultsConfigmap(ctx, c, configmapName, namespace)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("should return validation error when configmap is set to mutable", func() {
+		// Create a mutable ConfigMap
+		mutable := false
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      configmapName,
+				Namespace: namespace,
+			},
+			Data: map[string]string{
+				utils.UpgradeDefaultsConfigmapKey: `
+ibuSpec:
+  seedImageRef:
+    image: "quay.io/openshift-release-dev/ocp-release"
+    version: "4.17.0"
+  oadpContent:
+  - name: "oadp-backup"
+    namespace: "openshift-adp"
+plan:
+- actions: ["Prep"]
+- actions: ["Upgrade"]`,
+			},
+			Immutable: &mutable,
+		}
+		Expect(c.Create(ctx, cm)).To(Succeed())
+
+		err := t.validateUpgradeDefaultsConfigmap(ctx, c, configmapName, namespace)
+		Expect(err).To(HaveOccurred())
+		Expect(utils.IsInputError(err)).To(BeTrue())
+		Expect(err.Error()).To(Equal(fmt.Sprintf(
+			"It is not allowed to set Immutable to false in the ConfigMap %s", configmapName)))
+	})
+
+	It("should succeed when configmap is already immutable", func() {
+		// Create an already immutable ConfigMap
+		immutable := true
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      configmapName,
+				Namespace: namespace,
+			},
+			Data: map[string]string{
+				utils.UpgradeDefaultsConfigmapKey: `
+ibuSpec:
+  seedImageRef:
+    image: "quay.io/openshift-release-dev/ocp-release"
+    version: "4.17.0"
```

Comment:
nit: We should probably have updated version references in the new test code

Reply:
Will do in a follow-up PR